core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -1079,6 +1079,9 @@ class LogManager(logDirs: Seq[File],
       if (destLog == null)
         throw new KafkaStorageException(s"The future replica for $topicPartition is offline")
 
+      // the metrics tags still contain "future", so we have to remove it.
+      // we will add metrics back after sourceLog remove the metrics
+      destLog.removeLogMetrics()
       destLog.renameDir(UnifiedLog.logDirName(topicPartition), true)
       destLog.updateHighWatermark(sourceLog.highWatermark)
```

Comment:
Do you think we should remove log metrics after destLog got renamed? I'm thinking if we'll get stuck for some time during `renameDir` and it'll be helpful to have metrics reporting it. WDYT?

Reply:
>  it'll be helpful to have metrics reporting it

make sense. will copy that